modules/@shopify/checkout-sheet-kit/android/proguard-rules.pro

```diff
@@ -0,0 +1,12 @@
+# Keep the ObjectMapper and its methods
+-keep class com.fasterxml.jackson.databind.ObjectMapper { *; }
+
+# Keep Checkout Sheet Kit classes
+-keep class com.shopify.checkoutsheetkit.lifecycleevents.CheckoutCompletedEvent { *; }
+-keep class com.shopify.checkoutsheetkit.pixelevents.PixelEvent { *; }
+-keep class com.shopify.checkoutsheetkit.CheckoutException { *; }
+-keep class com.shopify.checkoutsheetkit.CheckoutExpiredException { *; }
+-keep class com.shopify.checkoutsheetkit.ClientException { *; }
+-keep class com.shopify.checkoutsheetkit.HttpException { *; }
+-keep class com.shopify.checkoutsheetkit.ConfigurationException { *; }
+-keep class com.shopify.checkoutsheetkit.CheckoutSheetKitException { *; }
```

Comment:
could/should this be added to the checkout-sheet-kit-android repo?

Reply:
Good question - do we have a rules file there already?

Reply:
This would make sense, looking at the current rules, I deduce the issue I faced is related to the underlying android lib being minified. If so, I assume my issue would also occur on a plain android app and then it would definitely make sense to drill up the proguard rules to the android lib.

Please note I didn't need to keep the fasterxml class, but I may have another lib doing it already, I didn't check

Reply:
@kiftio PR to add the rules to the Android kit https://github.com/Shopify/checkout-sheet-kit-android/pull/150

It's unclear to me whether we need to maintain the fasterxml class which is used in this kit. Let's ship the Android changes, upgrade the kit here and see if it works.